### PR TITLE
Fix apt repo issue for docker

### PR DESCRIPTION
**Pull request by ydcjeff**
## What does this PR do?

Fixes # (issue)

# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by ydcjeff** [comment]
Docker builds relating with nightly will fail at the moment since pytorch has released 1.7 rc version.

File: dockers/base-cuda/Dockerfile [hunk 1]
```diff
@@ -34,7 +34,7 @@ SHELL ["/bin/bash", "-c"]
 
 ENV PATH="$PATH:/root/.local/bin"
 
-RUN apt-get update && apt-get install -y --no-install-recommends \
+RUN apt-get clean && apt-get update && apt-get install -y --no-install-recommends \
```
Comment (Borda) [inline dockers/base-cuda/Dockerfile:37]: This seems strange, you need update before install not clean..

Comment (ydcjeff) [reply dockers/base-cuda/Dockerfile:37]: Turns out it's CDN issue, some suggest clean and update
but let's wait and see if CDN can solve this

https://github.com/NVIDIA/nvidia-docker/issues/1392
https://github.com/NVIDIA/nvidia-docker/issues/877
https://github.com/NVIDIA/nvidia-docker/issues/1328

File: dockers/base-xla/Dockerfile [hunk 2]
```diff
@@ -28,7 +28,7 @@ ENV CONDA_ENV=lightning
 # show system inforation
 RUN lsb_release -a && cat /etc/*-release
 
-RUN apt-get update && \
+RUN apt-get clean && apt-get update && \
```
Comment (Borda) [inline dockers/base-xla/Dockerfile:31]:
```suggestion
RUN apt-get update && \
```

File: dockers/base-cuda/Dockerfile (hunk 1 above)
Comment (Borda) [reply dockers/base-cuda/Dockerfile:37]: ok, the problem is in 
```
Reading package lists...
E: Failed to fetch https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64/Packages.gz  File has unexpected size (47871 != 49498). Mirror sync in progress? [IP: 152.195.19.142 443]
   Hashes of expected file:
    - Filesize:49498 [weak]
    - SHA256:332f3ee4e353b8a5e5a2bdd8fdbd47cf140c73822b82b328815f122e09e195a0
    - SHA1:4dc8ef9a3ee3c97b3c26d46e07fdd83997e6880b [weak]
    - MD5Sum:bbff3b9c3462257479d72521ee78ec29 [weak]
   Release file created at: Wed, 23 Sep 2020 22:09:13 +0000
E: Some index files failed to download. They have been ignored, or old ones used instead.
```
not the update step...

**Review by Borda — CHANGES_REQUESTED** [review]

File: dockers/base-cuda/Dockerfile (hunk 1 above)
Comment (ydcjeff) [reply dockers/base-cuda/Dockerfile:37]: That comes from this, I think
```
Get:10 https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64  Packages [49.5 kB]
Err:10 https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64  Packages
  File has unexpected size (47871 != 49498). Mirror sync in progress? [IP: 152.195.19.142 443]
  Hashes of expected file:
   - Filesize:49498 [weak]
   - SHA256:332f3ee4e353b8a5e5a2bdd8fdbd47cf140c73822b82b328815f122e09e195a0
   - SHA1:4dc8ef9a3ee3c97b3c26d46e07fdd83997e6880b [weak]
   - MD5Sum:bbff3b9c3462257479d72521ee78ec29 [weak]
  Release file created at: Wed, 23 Sep 2020 22:09:13 +0000
```

Comment (Borda) [reply dockers/base-cuda/Dockerfile]: so skip of the `update` is just hiding the NVIDIA package issue not solving it.. :[

**Review by Borda — APPROVED** [review]

**Comment by ydcjeff** [comment]
Is it okay to use CUDA 11?

**Review by justusschock — APPROVED** [review]

**Comment by ydcjeff** [comment]
Downgrading to ubuntu 16.04 works fine

**Review by SkafteNicki — APPROVED** [review]

**Review by Borda — CHANGES_REQUESTED** [review]
pls, no 5 years old OS

File: dockers/base-cuda/Dockerfile [hunk 3]
```diff
@@ -22,8 +22,11 @@
 ARG CUDNN_VERSION=7
 ARG CUDA_VERSION=10.1
 
-FROM nvidia/cuda:${CUDA_VERSION}-cudnn${CUDNN_VERSION}-devel
-# FROM nvidia/cuda:${CUDA_VERSION}-devel
+# FROM nvidia/cuda:${CUDA_VERSION}-devel-ubuntu20.04
+# FROM nvidia/cuda:${CUDA_VERSION}-cudnn${CUDNN_VERSION}-devel-ubuntu18.04
+FROM nvidia/cuda:${CUDA_VERSION}-cudnn${CUDNN_VERSION}-devel-ubuntu16.04
```
Comment (Borda) [inline dockers/base-cuda/Dockerfile:27]: this is bad, this is too. old OS version, this is the almost unsupported OS (ends in April 2021)

**Pull request merged**